hr_expense_invoice/models/hr_expense_expense.py

```diff
@@ -0,0 +1,103 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+
+from openerp import models, fields, api, exceptions, _
+
+
+class HrExpenseExpense(models.Model):
+    _inherit = 'hr.expense.expense'
+
+    @api.model
+    def line_get_convert(self, x, part, date):
+        res = super(HrExpenseExpense, self).line_get_convert(x, part, date)
+        if x.get('invoice'):
+            res['partner_id'] = (
+                x['invoice'].partner_id.commercial_partner_id.id)
```

Comment:
I'm afraid not, because sometimes I have faced that a positional argument is called from other overwritings with keyword arguments, and this renaming will break these calls, so we have to stick to these bad names. I found it renaming `partner` to `partner_id` argument on an onchange method (because semantically, it's an id, not the record itself).